lib/IgnorePlugin.js

```diff
@@ -8,14 +8,20 @@
 
 class IgnorePlugin {
 	/**
-	 * @param {RegExp} resourceRegExp A RegExp to test the request against
-	 * @param {RegExp=} contextRegExp A RegExp to test the context (directory) against
+	 * @param {object} options IgnorePlugin options
+	 * @param {RegExp} options.resourceRegExp - A RegExp to test the request against
+	 * @param {RegExp} options.contextRegExp - A RegExp to test the context (directory) against
+	 * @param {TODO} options.checkResource - A function that will take resource as argument and should return true or false
+	 * @param {TODO} options.checkContext - A function that will take context as argument and should return true or false
 	 */
-	constructor(resourceRegExp, contextRegExp) {
-		/** @private @type {RegExp} */
-		this.resourceRegExp = resourceRegExp;
-		/** @private @type {RegExp} */
-		this.contextRegExp = contextRegExp;
+	constructor(options) {
```

Comment:
add a schema like with other plugins (after the compat-layer)

Reply:
@sokra What do you mean by "after the compat-layer"? 
= Only for the webpack 5 version when we remove the compat-layer?

Reply:
`after` => after that code in the source file.

```
constructor(options) {
  // compat stuff
  // schema validation
  this.options = options;
}
```

Reply:
Ok I see, did it.